pkg/controller/pod_test.go

```diff
@@ -202,10 +202,72 @@ var _ = Describe("Dynamic Attachment controller", func() {
 					}).Should(BeEmpty())
 				})
 			})
+
+			When("an update to an existing attachment occurs", func() {
+				DescribeTable("nothing happens", func(updatedNetworkSelectionElement nad.NetworkSelectionElement) {
+					Expect(updatePodNetworkSelectionElements(k8sClient, pod, updatedNetworkSelectionElement)).To(Succeed())
+
+					_, err := k8sClient.CoreV1().Pods(namespace).UpdateStatus(
+						context.TODO(),
+						pod,
+						metav1.UpdateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					Consistently(func() ([]nad.NetworkStatus, error) {
+						updatedPod, err := k8sClient.CoreV1().Pods(namespace).Get(context.TODO(), podName, metav1.GetOptions{})
+						if err != nil {
+							return nil, err
+						}
+						status, err := networkStatus(updatedPod.Annotations)
+						if err != nil {
+							return nil, err
+						}
+						return status, nil
+					}).Should(ConsistOf(ifaceStatus(namespace, networkName, "net0", "")))
+				},
+					Entry("when the MAC address is updated", nad.NetworkSelectionElement{
+						Name:             networkName,
+						Namespace:        namespace,
+						InterfaceRequest: "net0",
+						MacRequest:       "07:06:05:04:03:02",
+					}),
+					XEntry("when the iface name is updated", nad.NetworkSelectionElement{
```

Comment:
This is wrong; the original attachment **does** feature the interface name. 

We need to somehow have a pod with a provisioned attachment that does not specify the interface name on the network selection elements.